lib/ecto_ldap/adapter.ex

```diff
@@ -230,9 +177,79 @@ defmodule Ecto.Ldap.Adapter do
   def translate_value(%Ecto.Query.Tagged{value: value}), do: value
   def translate_value(other), do: other
 
+  def execute(_repo, query_metadata, prepared, params, preprocess, _options) do
+
+    search_response   = search(prepared, params)
+    fields            = ordered_fields(query_metadata.sources)
+    count             = count_fields(query_metadata.select.fields, query_metadata.sources)
+
+    {:ok, {:eldap_search_result, results, []}} = search_response
+
+    result_set = results
+    |> Enum.map(&process_entry(&1))
+    |> Enum.map(&prune_attributes(&1, fields))
+    |> Enum.map(&generate_models(&1, preprocess, count))
```

Comment:
This area will need more attention.  I flipped the code over to use the `for` loop and it started complaining during postprocessing.  Definitely feels fixable once investigated in earnest.